packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -192,29 +197,55 @@ export class HttpPlugin extends BasePlugin<Http> {
         [AttributeNames.HTTP_HOSTNAME]: host,
         [AttributeNames.HTTP_METHOD]: method,
         [AttributeNames.HTTP_PATH]: options.path || '/',
-        [AttributeNames.HTTP_USER_AGENT]: userAgent || '',
       });
 
+      if (userAgent !== undefined) {
+        span.setAttribute(AttributeNames.HTTP_USER_AGENT, userAgent);
+      }
+      // In case of req.abort() is called after receiving response,
+      // req.on('close') is called before res.on('end')
+      let reqShouldCloseSpan: boolean = true;
       request.on(
         'response',
-        (response: IncomingMessage & { aborted?: boolean }) => {
+        (
+          response: IncomingMessage & { aborted?: boolean; req: ClientRequest }
+        ) => {
+          if (response.statusCode) {
+            span.setAttributes({
+              [AttributeNames.HTTP_STATUS_CODE]: response.statusCode,
+              [AttributeNames.HTTP_STATUS_TEXT]: response.statusMessage,
+            });
+          }
+          // Check if we have the original ClientRequest
+          // Otherwise it's a package that has a custom implementation
+          // custom implementation means that listeners func is not reliable (e.g superagent, request ...) provides their own way.
+          // and we can't avoid to listen to 'end' event even no 'data'/'readable' event is attached
+          if (response.req.constructor.name === 'ClientRequest') {
+            const dataFunc = response.listeners('data') as Function[];
```

Comment:
consider to use `response.listenerCount` to avoid unneeded creation of a temp array.